contract/src/RadicleCloudBase.sol

```diff
@@ -0,0 +1,264 @@
+    function _nextImpact() internal view returns (uint64) {
+        return min(min(buys.last(), expires.last()), min(cancels.last(), renews.last()));
+    }
+
+    /// @dev Calculate the amount operator has earned till current block.
+    /// @return revenue earned from last processed blocked till now
+    function _calcRealizedRevenue() internal returns (uint128) {
+        uint128 revenue = 0;
+        int64 rate = runningRate;
+        uint64 last = lastProcessed;
+        uint64 current = uint64(block.number);
+        while (last <= current) {
+            uint64 nextLast = _nextImpact();
+            if (nextLast == MAXBLOCK) break;
+            if (nextLast == last) {
+                rate += impacts[last];
+                if (last == buys.last()) buys.next();
+                if (last == cancels.last()) cancels.next();
+                if (last == renews.last()) renews.next();
+                if (last == expires.last()) {
+                    // create renews on fly
+                    if (hasRenewal[last]) {
+                        renews.add(last + duration);
+                        delete hasRenewal[last];
+                    }
+                    expires.next();
+                }
+                delete impacts[last];
+                nextLast = _nextImpact();
+            }
+            revenue += uint64(rate) * (min(current, nextLast) - last);
+            last = nextLast;
+        }
+
+        runningRate = rate;
+        lastProcessed = current;
+        return revenue;
+    }
+
+    /// @notice Update realized revenue and accounting for owner.
+    /// @return amount earned by owner till current block
+    function withdrawRealizedRevenue() public isOwner returns (uint128 amount) {
+        amount = _calcRealizedRevenue();
+        _withdraw(owner, amount);
+    }
+
+    /// @dev Cancel deployment.
+    /// @param org the org being cancelled
+    function _cancelDeploymentAndPayout(address org) internal {
+        // update impacts
+        impacts[uint64(block.number)] -= int64(dep[org].rate);
+        impacts[dep[org].expiry] += int64(dep[org].rate);
+        cancels.add(uint64(block.number));
+
+        // payout remaining balance
+        if (dep[org].expiry > block.number) {
+            uint128 leftBlocks = dep[org].expiry - uint128(block.number);
+            uint128 due = dep[org].rate * leftBlocks;
+            _withdraw(dep[org].owner, due);
+        }
+
+        emit DeploymentCancelled(org);
+    }
+
+    /// @notice Cancel service for org and withdraw remaining fund.
+    /// @param org the org being cancelled, must be owned by caller
+    function cancelService(address org) public {
+        require(dep[org].owner == msg.sender, "Caller does not own org");
+        require(
+            dep[org].start + withdrawWait <= block.number,
+            "Withdraw wait period hasn't finished"
+        );
+        _cancelDeploymentAndPayout(org);
```

Comment:
What's the difference between the service and the deployment? If there is none we should rename `DeploymentCancelled` and other uses of "deployment"

Reply:
https://github.com/radicle-dev/radicle-cloud/pull/1/commits/a70cc946d000375a8fb5c46bd48f6152789a0112